src/Splitter.sol

```diff
@@ -0,0 +1,114 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+interface VatLike {
+    function move(address, address, uint256) external;
+    function hope(address) external;
+    function nope(address) external;
+}
+
+interface DaiJoinLike {
+    function vat() external view returns (address);
+    function exit(address, uint256) external;
+}
+
+interface FlapLike {
+    function kick(uint256, uint256) external returns (uint256);
+    function cage(uint256) external;
+}
+
+interface FarmLike {
+    function notifyRewardAmount(uint256 reward) external;
+}
+
+contract Splitter {
+    mapping (address => uint256) public wards;
+    FlapLike    public           flapper;
+    uint256     public           burn; // in WAD; 1 WAD = 100% burn
+
+    VatLike     public immutable vat;
+    DaiJoinLike public immutable daiJoin;
+    FarmLike    public immutable farm;
+
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+    event File(bytes32 indexed what, address data);
+
+    constructor(
+        address _daiJoin,
+        address _farm
+    ) {
+        daiJoin = DaiJoinLike(_daiJoin);
+        vat     = VatLike(daiJoin.vat());
+        farm    = FarmLike(_farm);
+        
+        vat.hope(_daiJoin);
+        
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "Splitter/not-authorized");
+        _;
+    }
+
+    uint256 internal constant WAD = 10 ** 18;
+    uint256 internal constant RAY = 10 ** 27;
+
+    function rely(address usr) external auth { wards[usr] = 1; emit Rely(usr); }
+    function deny(address usr) external auth { wards[usr] = 0; emit Deny(usr); }
+
+    function file(bytes32 what, uint256 data) external auth {
+        if   (what == "burn") burn = data;
+        else revert("Splitter/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function file(bytes32 what, address data) external auth {
+        if (what == "flapper") {
+            vat.nope(address(flapper));
+            flapper = FlapLike(data);
+            vat.hope(data);
+        }
+        else revert("Splitter/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    // Note: the kick() cadence is determined by flapper.hop()
+    function kick(uint256 tot, uint256) external auth returns (uint256) {
+        vat.move(msg.sender, address(this), tot);
+
+        uint256 lot = tot * burn / WAD;
+        flapper.kick(lot, 0);
+
+        if (tot - lot >= RAY) {
+            FarmLike farm_ = farm;
+            uint256 pay = (tot - lot) / RAY;
```

Comment:
nit:
```suggestion
        uint256 pay = (tot - lot) / RAY;
        if (pay >= 0) {
            FarmLike farm_ = farm;
```

Reply:
I think dust can only accumulate here if `tot` is not a multiple of `RAD`, which seems like an unlikely setting. So I'd say it's fine to allow dust to theoretically accumulate for weird `bump` values that would not be multiples of `RAD`.

Reply:
A multiple of `RAY`, right?
It would also affect the value of `burn` to this, but yeah I guess it is not a big deal. Anyone could also send `vat.dai` and we would be in a similar situation. Pause proxy can actually remove this balance if needed as well.

Reply:
No I meant `RAD` so that this is independent of the `burn` value. If `tot = t * RAD` then `tot - tot * burn / WAD = t * RAD - t * RAD * burn / WAD = t * RAD - t * RAY * burn`, which is a multiple of `RAY` for any values of the `t` and `burn` integers.